Migrated OwnershipInfo to H5

BenGalewsky opened the pull request:
The ownershipInfo entity has one to many relationships with Asset and BeneficialOwner

In this change I changed the baseclass of OwnershipInfo to make the id more readable.

Had to change Asset and BeneficialOwner classes to reflect their manyToOne side of the relation.

Created a new table ins seed.sql

Eliminated the explicit Sequence

jasonaowen commented:
I've looked at this a few times, and reread [the JavaPersistance OneToMany chapter](https://en.wikibooks.org/wiki/Java_Persistence/OneToMany) once or twice, and I'm still not sure I understand.

- When are assets set? AFAICT, only in the data loader subproject, and I don't yet understand its relevance to the rest of the project (@slifty may know more soon). Can we just delete that field?
- If we have reciprocal relationships, both `@OneToMany` and `@ManyToOne`, how does the latter get set? This PR introduces getters and setters to both `Asset` and `BeneficialOwner`, but nothing ever calls either. The wikibook says:

  > Some expect the JPA provider to have magic that automatically maintains relationships. This was actually part of the EJB CMP 2 specification. However the issue is if the objects are detached or serialized to another VM, or new objects are related before being managed, or the object model is used outside the scope of JPA, then the magic is gone, and the application is left figuring things out, so in general it may be better to add the code to the object model. However some JPA providers do have support for automatically maintaining relationships.

  Does Hibernate do that automatically? Do we need to wire it up somehow? Or does the code that sets the `@OneToMany` relationship also need to set the `@ManyToOne` relationship?
- Have you been able to test this with `BeneficialOwner`? I do see code that sets the `@OneToMany` side, but I haven't traced through the code yet. Does it get called?

jasonaowen commented on psm-app/db/seed.sql:
```diff
@@ -1152,3 +1153,15 @@ CREATE TABLE provider_statements(
   "date" DATE
 );
+
+  CREATE TABLE ownership_info (
```
Please unindent.

jasonaowen commented on psm-app/services/src/main/java/gov/medicaid/entities/OwnershipInformation.java:
```diff
@@ -15,55 +15,71 @@
     /**
      * Beneficial owners.
      */
+    @OneToMany(fetch = FetchType.EAGER)
```
Why is `beneficialOwners` `EAGER` and `assets` is (default) `LAZY`?

BenGalewsky replied:
Mostly because we don't see Asset being used. There might be a performance impact of hitting the db for this empty table. I went ahead and made it consistent, and hopefully we can determine that the whole thing can be deleted.

BenGalewsky commented:
Thanks for checking this out @jasonaowen - your suspicion about the reflective ManyToOne was indeed correct. 
To verify:
1. I started an enrollment with Dental Clinic
2. Filled in Organization Info
3. Added an "Individual member" with a provider type of "Dentist"
4. On the ownership info page:
  - Entity type: Sole Proprietorship
  -  Added "Another Business Ownership or Control Interest"
  - Selected the checkbox for "This person/business has an ownership or control interest..."
  - Filled in the additional info boxes and checked "No" for the disclosures
5. Clicked on next
6. Clicked on "Save as Draft"

Go to the database and `select * from beneficial_owner;` to see that the beneficial owner record was created and that the ownership_info_id is not null and references the record in the ownership_info table.

When I tried this out with the OneToMany/ManyToOne setup I saw the record created, but ownership_info_id was null which makes sense since the code never sets the BeneficialOwner object's OwnerInfo property.
